lib/ansible/modules/system/launchd.py

```diff
@@ -0,0 +1,236 @@
+    - The plistlib python library
+'''
+
+EXAMPLES = '''
+- name: Make sure spotify webhelper is started
+  launchd:
+    name: com.spotify.webhelper
+    state: started
+    become: yes
+'''
+
+RETURN = '''
+status:
+    description: metadata about service status
+    returned: always
+    type: dict
+    sample:
+        {
+            "current_pid": "-",
+            "current_state": "stopped",
+            "previous_pid": "82636",
+            "previous_state": "running"
+        }
+'''
+
+import os
+from time import sleep
+
+try:
+    import plistlib
+    HAS_PLIST = True
+except ImportError:
+    HAS_PLIST = False
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def find_service_plist(service_name):
+    """Finds the plist file associated with a service"""
+
+    launchd_paths = [
+        '~/Library/LaunchAgents',
+        '/Library/LaunchAgents',
+        '/Library/LaunchDaemons',
+        '/System/Library/LaunchAgents',
+        '/System/Library/LaunchDaemons'
+    ]
+
+    for path in launchd_paths:
+        try:
+            files = os.listdir(os.path.expanduser(path))
+        except OSError:
+            continue
+
+        for filename in files:
+            if filename == '%s.plist' % service_name:
+                return os.path.join(path, filename)
+
+
+def main():
+    # init
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(
+                required=True,
+                aliases=['service']
+            ),
+            state=dict(
+                choices=['started', 'stopped', 'restarted', 'reloaded'],
+            ),
+            enabled=dict(type='bool'),
+            sleep=dict(type='int',
+                       default=1),
+        ),
+        supports_check_mode=True,
+        required_one_of=[
+            ['state', 'enabled']
+        ],
+    )
+
+    launch = module.get_bin_path('launchctl', True)
+    service = module.params['name']
+    action = module.params['state']
+    rc = 0
+    out = err = ''
+    result = {
+        'name': service,
+        'changed': False,
+        'status': {},
+    }
+
+    rc, out, err = module.run_command('%s list' % launch)
+    if rc != 0:
+        module.fail_json(msg='running %s list failed' % launch)
+
+    running = False
+    found = False
+
+    for line in out.splitlines():
+        if line.strip():
+            pid, last_exit_code, label = line.split('\t')
+            if label.strip() == service:
+                found = True
+                if pid != '-':
+                    running = True
+                    result['status']['previous_state'] = 'running'
+                    result['status']['previous_pid'] = pid
+                else:
+                    result['status']['previous_state'] = 'stopped'
+                    result['status']['previous_pid'] = '-'
+                break
+
+    if not found:
+        module.fail_json(msg="Unable to find the service %s among active services." % service)
```

Comment:
For debugging purposes it would help if `out` would be dumped.
Example:
```
        msg = "Unable to find the service %s among active services." % service
        if module._verbosity >= 1:
          msg += "\r\nRunning services:\r\n%s" % out
        module.fail_json(msg=msg)
```